### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by rahimialijan**
# OOP school library: add basic UI
In this project, I created a form of UI for my school library app. This way it can be invoked as an executable and not something I use in IRB exclusively.

1. Created an app.rb file and used different methods for that.
2. Created a main.rb file to define the entry point, and used different methods also.
3. Resolve the linter errors

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    if person.nil?
+      puts "Person with ID #{person_id} not found."
+      return
+    end
+
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+
+    puts "Rental of '#{book.title}' by #{book.author} to #{person.correct_name} on #{rental.date} has been created."
+  end
```
Comment (julie-ify) [inline app.rb]: - [ ] The app crashes when I try to create a rental. Kindly inspect and fix that.

  image

File: main.rb [hunk 2]
```diff
@@ -1,11 +1,88 @@
...
+def create_rental(app)
+  app.create_rental
+end
+
+def list_rentals(app)
+  print 'Enter person ID: '
+  person_id = gets.chomp.to_i
+  app.list_rentals_for_person(person_id)
+end
+
```
Comment (julie-ify) [inline main.rb]: - [ ] The app logic should be in the `app.rb` file. The purpose of the `main.rb` is solely for presenting the initial options to the user. Kindly move these methods to the app file.

File: main.rb [hunk 3]
```diff
@@ -1,11 +1,88 @@
...
+    when 2
+      app.list_people
+    when 3
+      create_person(app)
+    when 4
+      create_book(app)
+    when 5
+      create_rental(app)
+    when 6
+      list_rentals(app)
```
Comment (julie-ify) [inline main.rb:40]: - [ ] As a result of my being unable to create a rental I could not test the list rental option. Kindly ensure this option works too.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @rahimialijan,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (rahimialijan) [reply app.rb]: Thanks for you reviewing my PR, But it correctly works on my system. I will try my best and solve the issue

File: main.rb (hunk 2 above)
Comment (rahimialijan) [reply main.rb]: done

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  print 'Enter book title: '
+  title = gets.chomp
+  print 'Enter book author: '
+  author = gets.chomp
+  app.create_book(title, author)
+end
+
+def create_rental(app)
+  app.create_rental
+end
```
Comment (nahnahdev) [inline app.rb:109]: - [ ] Great job so far @rahimialijan However, just as the previous reviewer mentioned, I am still unable to create a rental. I am also unable to test the `List rentals for a person` option as well. So kindly fix the errors and ensure that both options work as expected :wink: 
- Error Message
image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @rahimialijan  , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights :fire: 
- Most requested changes implemented :heavy_check_mark: 
- Linters well setup and passing :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @rahimialijan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good requested changes implementation ✔️ 
- Good UI implementation ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**